src/NServiceBus.Core/Transports/Msmq/MsmqUtilities.cs

```diff
@@ -71,6 +71,34 @@ static MsmqAddress GetIndependentAddressForQueue(MessageQueue q)
             return headers;
         }
 
+        public static bool TryOpenQueue(string address, out MessageQueue messageQueue)
+        {
+            messageQueue = null;
+            var msmqAddress = MsmqAddress.Parse(address);
+            var path = msmqAddress.PathWithoutPrefix;
+            try
+            {
+                if (MessageQueue.Exists(path))
+                {
+                    messageQueue = new MessageQueue(path);
+                    return true;
```

Comment:
Instead of having this `return true` statement and the `return false` statement at the end of the method, you could remove this and change the return statement at the end to be: `return messageQueue != null`. That would allow for a single return statement in the method, which I tend to prefer when possible.